File: php/features.php
<?php
declare(strict_types=1);

// DataUsa SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DataUsaFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DataUsaBaseFeature();
            case "test":
                return new DataUsaTestFeature();
            default:
                return new DataUsaBaseFeature();
        }
    }
}

File: py/features.py
# DataUsa SDK feature factory

from feature.base_feature import DataUsaBaseFeature
from feature.test_feature import DataUsaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DataUsaBaseFeature(),
        "test": lambda: DataUsaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
